range.go

```diff
@@ -197,3 +197,28 @@ func (m rangeOpMatcher) String() string {
 		m.eqScanOp.String(),
 	)
 }
+
+// IndexFromConditions returns the name of the index or the base table to use, along with the key info
+// for that index. If no suitable index could be found, an error is returned
+func (ei *EntityInfo) IndexFromConditions(conditions map[string][]*Condition) (name string, key *PrimaryKey, err error) {
```

Comment:
nit: This function name seems a little misleading to me, since it doesn't necessarily return an Index in the conventional sense. Not sure what would be better though... :/